### ISSUE TITLE
Concurrent ComputeFundamentalMatrix fails

### ISSUE THREAD

**Issue by jessemapel** [USGS-Astrogeology/autocnet#305]
When using `cng.apply(ComputeFundamentalMatrix, ...)`, there are failures due to concurrent updates in the `validate_points` trigger.

Sample output log for a failed job

```
/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/sql/functions.py:68: SAWarning: The GenericFunction 'array_agg' is already registered and is going to be overriden.
  "is going to be overriden.".format(identifier))
Traceback (most recent call last):
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1244, in _execute_context
    cursor, statement, parameters, context
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 550, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.InternalError_: tuple concurrently updated


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jmapel/.conda/envs/autocnet/bin/acn_submit", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/work/users/jmapel/autocnet/bin/acn_submit", line 11, in <module>
    from autocnet.io.db import connection
  File "/work/users/jmapel/autocnet/autocnet/__init__.py", line 52, in <module>
    import autocnet.graph
  File "/work/users/jmapel/autocnet/autocnet/graph/__init__.py", line 1, in <module>
    from . import edge
  File "/work/users/jmapel/autocnet/autocnet/graph/edge.py", line 15, in <module>
    from autocnet.graph.node import Node
  File "/work/users/jmapel/autocnet/autocnet/graph/node.py", line 20, in <module>
    from autocnet.io.db.model import Images, Keypoints, Matches, Cameras,  Base, Overlay, Edges, Costs, Points, Measures
  File "/work/users/jmapel/autocnet/autocnet/io/db/model.py", line 277, in <module>
    Keypoints.__table__])
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", line 4287, in create_all
    ddl.SchemaGenerator, self, checkfirst=checkfirst, tables=tables
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 2033, in _run_visitor
    conn._run_visitor(visitorcallable, element, **kwargs)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1607, in _run_visitor
    visitorcallable(self.dialect, self, **kwargs).traverse_single(element)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", line 131, in traverse_single
    return meth(obj, **kw)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", line 769, in visit_metadata
    _ddl_runner=self,
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/event/attr.py", line 297, in __call__
    fn(*args, **kw)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", line 255, in __call__
    return bind.execute(self.against(target))
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 988, in execute
    return meth(self, multiparams, params)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", line 72, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1050, in _execute_ddl
    compiled,
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1248, in _execute_context
    e, statement, parameters, cursor, context
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1466, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 383, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 128, in reraise
    raise value.with_traceback(tb)
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1244, in _execute_context
    cursor, statement, parameters, context
  File "/home/jmapel/.conda/envs/autocnet/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 550, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.InternalError: (psycopg2.errors.InternalError_) tuple concurrently updated

[SQL: 
CREATE OR REPLACE FUNCTION validate_points()
  RETURNS trigger AS
$BODY$
BEGIN
 IF (SELECT COUNT(*) 
	 FROM MEASURES 
	 WHERE pointid = NEW.pointid AND active = True) < 2
 THEN 
   UPDATE points
     SET active = False
	 WHERE points.id = NEW.pointid;
 ELSE
   UPDATE points
   SET active = True
   WHERE points.id = NEW.pointid;
 END IF;
 
 RETURN NEW;
END;
$BODY$

LANGUAGE plpgsql VOLATILE -- Says the function is implemented in the plpgsql language; VOLATILE says the function has side effects.
COST 100; -- Estimated execution cost of the function.
]
(Background on this error at: http://sqlalche.me/e/2j85)
```

**Comment by jessemapel** [comment]
For now a work around is to use `cng.compute_fundamental_matrices(...)` as this does not have the same problem

**Comment by jlaura** [comment]
It looks like the issue is that, on import, the code attempts to register a function with the DB. When many processes do this concurrently. Things eat it. You learn something every day! 

I believe that [this](is the offending code).

```
    # Trigger that watches for points that should be active/inactive
    # based on the point count.
    event.listen(Base.metadata, 'before_create', valid_point_function)
    event.listen(Measures.__table__, 'after_create', valid_point_trigger)
```

**Comment by jlaura** [comment]
@jessemapel Can you test: https://github.com/USGS-Astrogeology/autocnet/pull/306 ?

If it looks good we can merge and close.

**Comment by jessemapel** [comment]
I'm running Kelvin's region in serial right now, but @kaitlyndlee can test this.

**Comment by kaitlyndlee** [comment]
Fixes the problem, but a new error came up. Looks like the same error Kelvin and Jesse are running into.
```
/work/users/kdlee/anaconda3/envs/autocnet_local/lib/python3.6/site-packages/sqlalchemy/sql/functions.py:68: SAWarning: The GenericFunction 'array_agg' is already registered and is going to be overriden.
  "is going to be overriden.".format(identifier))
/home/kdlee/builds/autocnet_new/autocnet/bin/acn_submit:19: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  config = yaml.load(f)
Traceback (most recent call last):
  File "/work/users/kdlee/anaconda3/envs/autocnet_local/bin/acn_submit", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/kdlee/builds/autocnet_new/autocnet/bin/acn_submit", line 68, in <module>
    response = main(msg, config)
  File "/home/kdlee/builds/autocnet_new/autocnet/bin/acn_submit", line 45, in main
    res = func(*msg['args'], **msg['kwargs'])
  File "/home/kdlee/builds/autocnet_new/autocnet/autocnet/graph/edge.py", line 347, in compute_fundamental_matrix
    s_keypoints, d_keypoints = self.get_match_coordinates(clean_keys=clean_keys)
  File "/home/kdlee/builds/autocnet_new/autocnet/autocnet/graph/edge.py", line 704, in get_match_coordinates
    skps = matches[['source_x', 'source_y']].astype(np.float)
  File "/work/users/kdlee/anaconda3/envs/autocnet_local/lib/python3.6/site-packages/pandas/core/frame.py", line 2934, in __getitem__
    raise_missing=True)
  File "/work/users/kdlee/anaconda3/envs/autocnet_local/lib/python3.6/site-packages/pandas/core/indexing.py", line 1354, in _convert_to_indexer
    return self._get_listlike_indexer(obj, axis, **kwargs)[1]
  File "/work/users/kdlee/anaconda3/envs/autocnet_local/lib/python3.6/site-packages/pandas/core/indexing.py", line 1161, in _get_listlike_indexer
    raise_missing=raise_missing)
  File "/work/users/kdlee/anaconda3/envs/autocnet_local/lib/python3.6/site-packages/pandas/core/indexing.py", line 1246, in _validate_read_indexer
    key=key, axis=self.obj._get_axis_name(axis)))
KeyError: "None of [Index(['source_x', 'source_y'], dtype='object')] are in the [columns]"
```
Should I merge the PR still?

**Comment by jlaura** [comment]
Closed by #306.
